### PR TITLE
Feature - recovered status

### PR DESCRIPTION
**Pull request by joefitter**
* Added recovered status, only usable by asru admin, autoforwards to with-inspectorate
* Dont include recovered in next steps - (decorator)

### CODE REVIEW

File: lib/flow/get-next-steps-for-user.js [hunk 1]
```diff
@@ -77,6 +77,10 @@ module.exports = (task, profile) => {
         nextSteps.push(discardedByAsru);
       }
 
+      if (!taskIsOpen(task) && userIsAsruAdmin(profile)) {
```
Comment (wheelsandcogs) [inline lib/flow/get-next-steps-for-user.js:80]: Does this mean we can recover a granted task as well? Do we specifically want to check for `rejected` here?

File: lib/flow/status.js [hunk 2]
```diff
@@ -139,6 +143,12 @@ module.exports = {
     id: 'rejected',
     commentRequired: true
   },
+  recovered: {
+    id: 'recovered',
+    commentRequired: false,
```
Comment (wheelsandcogs) [inline lib/flow/status.js:148]: Should we enforce an explanation, or do we think it's not necessary?

File: lib/flow/get-next-steps-for-user.js (hunk 1 above)
Comment (joefitter) [reply lib/flow/get-next-steps-for-user.js:80]: you're right - it does in client but it also wants to be more specific here - I'll update

File: lib/flow/status.js (hunk 2 above)
Comment (joefitter) [reply lib/flow/status.js:148]: it is a button style form with a js.confirm dialog, I think that's fine as it's only going to realistically be done by Martin

File: lib/decorators/next-steps.js [hunk 3]
```diff
@@ -11,6 +11,9 @@ module.exports = settings => {
 
     return getNextStepsForUser(c, user.profile)
       .then(nextSteps => {
+        // recovered should not appear in next steps radios
+        nextSteps = nextSteps.filter(s => s.id !== recovered.id);
```
Comment (wheelsandcogs) [inline lib/decorators/next-steps.js:15]: We don't do this for `discardedByAsru` which is added to the available steps in the same way and displays a similar UI - do we need it?

Comment (joefitter) [reply lib/decorators/next-steps.js:15]: yes or no, I do not know how that works

**Pull request merged**